### DISCUSSION TITLE
Microcks Server Installation Error - Invalid parameter: redirect_uri

### DISCUSSION THREAD

**Discussion by cvsudheer108** [microcks/microcks#917] [Q&A]
Hello,
I am installing Microcks on AWS EC2.
I made the following modifications:
**docker-compose.yml:**
* Replaced localhost in the following entries with the respective IP:
KC_HOSTNAME_ADMIN_URL: "http://localhost:18080"
KC_HOSTNAME_URL: "http://localhost:18080"
KEYCLOAK_PUBLIC_URL=http://localhost:18080

* Whitelisted the ports : 8080 and 18080 (for keycloak) on EC2 security group

install/keycloak-realm
* In, microcks-realm-sample.json, replaced localhost with IP:
 "redirectUris": [
        "http://localhost:8080/*"
      ]

* Restarted docker compose

But still it fails with the above error.
Please advise.

**Comment by cvsudheer108** [comment]
<img width="1437" alt="Screenshot 2023-09-19 at 11 14 58 PM" src="https://github.com/microcks/microcks/assets/22774621/91e8e6eb-41c2-4d24-aebe-ac59448cff42">
THis is the screenshot of the error

**Reply by lbroudoux** [reply]
Hey there! Have you been able to log in to the Keycloak console to check the configuration of the `microcks` realm? At first sight, your configuration looks ok...

**Reply by cvsudheer108** [reply]
No.. I haven't checked that. Can you please guide me to some documentation or steps to check on that, please?

**Reply by lbroudoux** [reply]
Using your browser, you must connect to `KEYCLOAK_PUBLIC_URL` and log in using `admin` / `admin` (the default password that must be changed). After that, select the `microcks` realm and check the configuration of the `microcks-app-js` client. You should have the redirect URLs correctly setup.

**Reply by cvsudheer108** [reply]
Thanks!
The console UI is loading fine on the server, but when I click on : Administration Console link, it gave error that SSL is rquired.

<img width="825" alt="Screenshot 2023-09-20 at 1 23 35 PM" src="https://github.com/microcks/microcks/assets/22774621/9a8e4900-e7af-4db1-aaf4-eb19fdcf59b5">

I came across this documentation for Keycloak for configuring for Production. I am trying that. https://www.keycloak.org/server/configuration-production

Meanwhile, I came across this article on setting up Microcks on AWS EC2.
https://microcks.io/blog/install-microcks-on-aws/
It mentions that I just need to run a script - **Setup-microcks-apt.sh** - , but I couldn't find it in the repo.
Please let me know if there is a latest version of the same.

**Reply by cvsudheer108** [reply]
Up on further checking, I found that, I had missed the IP address in : microcks-realm-sample.json. After replacing there "localhost" with myIP address, the Microcks website strated working in HTTP. But the Keycloaks Admin console still failed to load with the HTTPS error.
I imported a Postman colelction that worked on my localhost machine.
The importing worked, but the API is not visible under the : "API/Services" tab.
On checking the microcks logs, found that the import is successful, but the service load is not getting listed.. logs below..

11:37:05.235 DEBUG 1 --- [    task-5] i.g.microcks.service.MessageService      : Emitting 4 request/response pair(s) as result
11:37:05.254 DEBUG 1 --- [    task-5] i.g.microcks.service.MessageService      : Found 3 request(s) for operation 650ac881e88bb12824162125-GET /energy/v1/switch-status
11:37:05.254 DEBUG 1 --- [    task-5] i.g.microcks.service.MessageService      : Found 3 response(s) for operation 650ac881e88bb12824162125-GET /energy/v1/switch-status
11:37:05.254 DEBUG 1 --- [    task-5] i.g.microcks.service.MessageService      : Emitting 3 request/response pair(s) as result
11:37:05.261 DEBUG 1 --- [    task-5] i.g.m.l.TestConformanceMetricConfigurer  : Processing of ServiceChangeEvent done !
11:37:10.822  WARN 1 --- [080-exec-2] o.k.adapters.RequestAuthenticator        : SSL is required to authenticate. Remote address 49.249.89.250 is secure: false, SSL required for: EXTERNAL .
11:37:10.828  WARN 1 --- [080-exec-8] o.k.adapters.RequestAuthenticator        : SSL is required to authenticate. Remote address 49.249.89.250 is secure: false, SSL required for: EXTERNAL . 

Might be it means that I need to have SSL, if I am using IP address.. but just wanted to convey this issue...

**Reply by lbroudoux** [reply]
Yeah ... this blog https://microcks.io/blog/install-microcks-on-aws/ is outdated and we changed so many things since its write-up. 
Your issue is still a problem with SSL/TLS protection of Keycloak that's now required. Typically we used to have a `setup-tls-run.sh` script that was generating self-signed certificates for both Microcks and Keycloak but we dropped them at Microcks `1.2.0`. There are still in repo on `1.1.0` tag though if you'd like some inspiration: https://github.com/microcks/microcks/tree/1.1.0/install/docker-compose
